### ISSUE TITLE
aws-dynamodb: Disabling dynamodb.Table timeToLiveAttribute results in failing deployment

### ISSUE THREAD

**Issue by alharris-at** [aws/aws-cdk#25655]
### Describe the bug

When using the `timeToLiveAttribute` attribute on the `Table` construct for dynamodb, it cannot be disabled or removed. This seems to be new behavior. It's not clear if this is a CDK or Service change that may be causing the issue.

Remove the attribute results in the following error message:
```
Resource handler returned message: "Invalid request provided: AttributeName property of TimeToLiveSpecification is required when TTL status is enabled or while enabling/disabling TTL." (RequestToken: <TOKEN>, HandlerErrorCode: InvalidRequest)
```

### Expected Behavior

I can remove a TTL from a ddb table via CDK.

### Current Behavior

An error is thrown.

### Reproduction Steps

```typescript
#!/usr/bin/env node
import 'source-map-support/register';
import { App, Stack } from 'aws-cdk-lib';
import { AttributeType, Table } from 'aws-cdk-lib/aws-dynamodb';

const app = new App();
const stack = new Stack(app, 'ReproStack');

new Table(stack, 'ReproTable', {
  partitionKey: { name: 'id', type: AttributeType.STRING },
  // timeToLiveAttribute: '_ttl', // Toggle this property on/off to cause failure
});
```

### CDK CLI Version

2.79.1 (build 2e7f8b7)

### Node.js Version

v18.15.0

### OS

Mac OSX

### Language

Typescript

### Language Version

Typescript 5.0.4

### Other information

{
  "name": "ddb_repro",
  "version": "0.1.0",
  "bin": {
    "ddb_repro": "bin/ddb_repro.js"
  },
  "scripts": {
    "build": "tsc",
    "watch": "tsc -w",
    "test": "echo 'NO TESTS'",
    "cdk": "cdk"
  },
  "devDependencies": {
    "@types/node": "20.1.0",
    "aws-cdk": "2.79.1",
    "ts-node": "^10.9.1",
    "typescript": "~5.0.4"
  },
  "dependencies": {
    "aws-cdk-lib": "2.79.1",
    "constructs": "^10.0.0",
    "source-map-support": "^0.5.21"
  }
}

I created a minimal repo with the error here https://github.com/alharris-at/ddbRepro

**Comment by TheRealAmazonKendra** [comment]
This appears to be a duplicate of https://github.com/aws/aws-cdk/issues/13106.

**Comment by TheRealAmazonKendra** [comment]
I'm going to close this since it is a duplicate of the ticket I linked above. If you need to immediately unblock yourself, you can use an escape hatch to set the `enabled` property on the underlying L1 to `false`. See https://docs.aws.amazon.com/cdk/v2/guide/cfn_layer.html#cfn_layer_raw.

**Comment by alharris-at** [comment]
Hi @TheRealAmazonKendra - I don't believe this is exactly the same as that issue (or it's related, but is in fact a new regression) In Amplify we've had E2E tests which verify the ability to toggle L2 DDB `timeToLiveAttribute` between being set and disabled, and it appears to have broken some time in the last week on the construct.

I have a passing e2e test that as of 5/17 demonstrating this functionality working - [CircleCI Test Artifacts](https://output.circle-artifacts.com/output/job/3740f953-d260-454b-b56c-9d2b09d7adb1/artifacts/0/packages/amplify-e2e-tests/amplify-e2e-reports/index.html) _You can refer to the test called `init a project with conflict detection enabled and toggle`_.

I don't think this is actually an issue in the CDK library, since it seems to affect older versions (which used to correctly support this), so I suspect a DDB Control Plane change of some sort triggered this.

The only reason I bring this up is that you may see customers (Like Amplify and our users) which are affected by this regression in DDB, and may want a more specific issue to +1 or reference than https://github.com/aws/aws-cdk/issues/13106.
